dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java

```diff
@@ -435,25 +435,34 @@ public User removeUserFromGroup(@ApiParam(hidden = true) @Auth User authUser,
         return FluentIterable.from(byId.getEntries()).filter(Tool.class).toList();
     }
 
-
     @GET
     @Timed
     @UnitOfWork
     @Path("/starredTools")
     @ApiOperation(value = "Get the logged-in user's starred tools", response = Entry.class, responseContainer = "List")
-    public Set<Entry> getStarredTools(@ApiParam(hidden = true) @Auth User user) {
+    public SortedSet<Entry> getStarredTools(@ApiParam(hidden = true) @Auth User user) {
         User u = userDAO.findById(user.getId());
-        return u.getStarredEntries().stream().filter(element -> element instanceof Tool).collect(Collectors.toSet());
+        return filter(u.getStarredEntries(), Tool.class);
     }
 
     @GET
     @Timed
     @UnitOfWork
     @Path("/starredWorkflows")
     @ApiOperation(value = "Get the logged-in user's starred workflows", response = Entry.class, responseContainer = "List")
-    public Set<Entry> getStarredWorkflows(@ApiParam(hidden = true) @Auth User user) {
+    public SortedSet<Entry> getStarredWorkflows(@ApiParam(hidden = true) @Auth User user) {
         User u = userDAO.findById(user.getId());
-        return u.getStarredEntries().stream().filter(element -> element instanceof Workflow).collect(Collectors.toSet());
+        return filter(u.getStarredEntries(), Workflow.class);
+    }
+
+    private SortedSet<Entry> filter(SortedSet<Entry> set, Class someClass) {
+        SortedSet filteredSet = new TreeSet();
+        for (Entry entry : set) {
```

Comment:
FYI, the bigger picture here is that the users objects are picked up by the DAO from the database in sorted order due to the order by annotation. They're stored in a SortedSet (which may or may not sort it again depending on the implementation) and then when you make this call, you're creating a new sorted set yet again.  

Just store the sorted entries from the DB in a list. And when filtering, just put the results in a list to avoid sorting it again. And test it to ensure that the database sorting is correct. Thanks
Look up the